File: examples/http_server.js
// HTTP server — Node.js on MerlionOS.
// Run: run-user node http_server.js

const http = require("http");

const server = http.createServer((req, res) => {
    if (req.url === "/") {
        res.writeHead(200, { "Content-Type": "text/plain" });
        res.end("Hello from Node.js on MerlionOS!\n");
    } else if (req.url === "/status") {
        res.writeHead(200, { "Content-Type": "application/json" });
        res.end(JSON.stringify({
            os: "MerlionOS",
            runtime: "Node.js",
            version: process.version,
            pid: process.pid,
            uptime: process.uptime(),
        }, null, 2) + "\n");
    } else {
        res.writeHead(404);
        res.end("Not Found\n");
    }
});

server.listen(8080, "0.0.0.0", () => {
    console.log("Node.js HTTP server on :8080");
});

File: examples/http_server.py
#!/usr/bin/env python3
"""Simple HTTP server — Python on MerlionOS."""

from http.server import HTTPServer, BaseHTTPRequestHandler
import json
import os
import sys

class MerlionHandler(BaseHTTPRequestHandler):
    def do_GET(self):
        if self.path == "/":
            body = "Hello from Python HTTP server on MerlionOS!\n"
            self.send_response(200)
            self.send_header("Content-Type", "text/plain")
            self.end_headers()
            self.wfile.write(body.encode())

        elif self.path == "/status":
            status = {
                "os": "MerlionOS",
                "python": sys.version,
                "pid": os.getpid(),
            }
            body = json.dumps(status, indent=2) + "\n"
            self.send_response(200)
            self.send_header("Content-Type", "application/json")
            self.end_headers()
            self.wfile.write(body.encode())

        elif self.path == "/health":
            self.send_response(200)
            self.end_headers()
            self.wfile.write(b"OK\n")

        else:
            self.send_response(404)
            self.end_headers()
            self.wfile.write(b"Not Found\n")

    def log_message(self, format, *args):
        print(f"[python-http] {args[0]}")

if __name__ == "__main__":
    port = 8080
    server = HTTPServer(("0.0.0.0", port), MerlionHandler)
    print(f"Python HTTP server on :{port}")
    server.serve_forever()
